src/components/CopyToClipboard/CopyToClipboard.tsx

```diff
@@ -2,77 +2,41 @@ import React from 'react';
 
 import ReactCopyToClipboard from 'react-copy-to-clipboard';
 
-import type {
-    CopyToClipboardBaseProps,
-    CopyToClipboardContent,
-    CopyToClipboardStatus,
-} from './types';
+import type {CopyToClipboardProps, CopyToClipboardStatus} from './types';
 
-interface CopyToClipboardGeneralProps extends CopyToClipboardBaseProps {
-    children: CopyToClipboardContent;
-}
-
-interface CopyToClipboardDefaultProps {
-    timeout: number;
-}
+const INITIAL_STATUS: CopyToClipboardStatus = 'pending';
 
-interface CopyToClipboardInnerProps
-    extends CopyToClipboardGeneralProps,
-        CopyToClipboardDefaultProps {}
+export function CopyToClipboard(props: CopyToClipboardProps) {
+    const {children, text, options, timeout, onCopy} = props;
 
-export interface CopyToClipboardProps
-    extends CopyToClipboardGeneralProps,
-        Partial<CopyToClipboardDefaultProps> {}
+    const [status, setStatus] = React.useState<CopyToClipboardStatus>(INITIAL_STATUS);
 
-interface CopyToClipboardState {
-    status: CopyToClipboardStatus;
-}
+    const timerIdRef = React.useRef<number>();
 
-export class CopyToClipboard extends React.Component<
-    CopyToClipboardInnerProps,
-    CopyToClipboardState
-> {
-    static INITIAL_STATUS: CopyToClipboardStatus = 'pending';
+    const content = React.useMemo(() => children(status), [children, status]);
 
-    state: CopyToClipboardState = {
-        status: CopyToClipboard.INITIAL_STATUS,
-    };
+    const handleCopy = React.useCallback<Required<ReactCopyToClipboard.Props>['onCopy']>(
```

Comment:
Do we really need `useCallback` here?

Reply:
I prefer to pass memoized object and fn values to React elements (non native HTML elements)